llvm/lib/Target/RISCV/RISCVGatherScatterLowering.cpp

```diff
@@ -349,6 +349,22 @@ RISCVGatherScatterLowering::determineBaseAndStride(Instruction *Ptr,
 
   SmallVector<Value *, 2> Ops(GEP->operands());
 
+  // If the base pointer is a vector, check if it's strided.
+  if (GEP->getPointerOperand()->getType()->isVectorTy()) {
+    auto [BaseBase, Stride] = determineBaseAndStride(
+        cast<Instruction>(GEP->getPointerOperand()), Builder);
```

Comment:
this might be pedantic, but where did we check if the pointer operand is an instruction.

Reply:
Whoops, we didn't